src/main/java/com/example/springBoot_hibernate_ferrocarriles/controller/ReportesController.java

```diff
@@ -0,0 +1,94 @@
+package com.example.springBoot_hibernate_ferrocarriles.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import com.example.springBoot_hibernate_ferrocarriles.model.EquipoDeTraccion;
+import com.example.springBoot_hibernate_ferrocarriles.model.Itinerario;
+import com.example.springBoot_hibernate_ferrocarriles.model.Locomotora;
+import com.example.springBoot_hibernate_ferrocarriles.service.EquipoDeTraccionService;
+import com.example.springBoot_hibernate_ferrocarriles.service.ItinerarioService;
+import com.example.springBoot_hibernate_ferrocarriles.service.LocomotoraService;
+
+@RestController
+public class ReportesController {
+	@Autowired
+	EquipoDeTraccionService equipoDeTraccionService;
+	@Autowired
+	ItinerarioService itinerarioService;
+	@Autowired 
+	LocomotoraService locomotoraService;
+	List<String> listaAuxiliar = new ArrayList<>();
+
+	public ReportesController() {
+		super();
+	}
+	
+	@GetMapping("/reportes/equiposAptos/{idItinerario}")
+	public List<EquipoDeTraccion> equiposAptosParaItinerario(@PathVariable("idItinerario")int idItinerario){
+		List<EquipoDeTraccion> equiposAptosParaItinerario = new ArrayList<>();
+		for(EquipoDeTraccion equipo:equipoDeTraccionService.getAllEquipoDeTraccion()) {
+			if(equipo.cantidadMaxCoches()>=itinerarioService.getItinerarioById((long) idItinerario).getCantidadVagones() && equipo.getLineaDeTrenes()==1) {
+				equiposAptosParaItinerario.add(equipo);
+			}
+		}		
+		return equiposAptosParaItinerario;
+	}
+	
+	@GetMapping("/reportes/locomotorasDeMarca/{marca}")
+	public List<Locomotora> locomotorasDeMarca(@PathVariable("marca")String marca){
+		List<Locomotora> locomotorasDeMarca = new ArrayList<>();
+		for(Locomotora locomotora:locomotoraService.getAllLocomotora()) {
```

Comment:
Controller classes shouldn't have logic inside, they rely on Services to perform business logic. So you should have a ReportService class to delegate these functionalities.